pydantic/_internal/_generate_schema.py

```diff
@@ -204,14 +187,12 @@ def model_schema(self, cls: type[BaseModel]) -> core_schema.CoreSchema:
         """
         Generate schema for a pydantic model.
         """
-        from pydantic.main import BaseModel
+        model_ref, schema = self._get_or_cache_recursive_ref(cls)
```

Comment:
we should probably move this down and make it private, it's not used outside the class afaik.

Reply:
From `main.py` on `BaseModel`:
```
    @classmethod
    def __get_pydantic_core_schema__(cls, source: type[BaseModel], gen_schema: GenerateSchema) -> CoreSchema:
        return gen_schema.model_schema(cls)
```

Reply:
Was able to find a way to make this private; done